### DISCUSSION TITLE
Set matching, Difference Mode questions and more

### DISCUSSION THREAD

**Discussion by StoneyBricks** [rgriebl/brickstore#377] [Q&A]
I am relatively new to the Bricklink world and have discovered your great software tool to make many things easier.
I have some questions about BrickStore:

- On your webpage, it says that there is a "Set Matching" function. Where can I find that and how does it work?
- What is "Difference Mode". what does that do, how to set values?

- I also have a well discussed question: how can load in a "want list", find the items on my own inventory list by using some kind of comparison or find function....and ultimately make inventory Quantity adjustments in a more automated method.?

Thanks,
John
Stoney Bricks Canada

**Comment by gwijnja** [comment]
I'm having trouble finding the "set matching" feature as well. If I select multiple parts in the inventory, it does fill the "appears in sets" sidebar, but I don't think that is the set matching feature, is it?

The website specifically says:

> given a list of items, BrickStore can tell you which sets you could build from them.

And I feel like I have looked through the whole application, but I can't find it.

**Comment by StoneyBricks** [comment]
Hi

Once you add a bunch of items to the list in Brickstore, you need to select them all. Ctlr- A. Then the list of sets on the left hand side will be narrowed down to those sets that contain all the selected items.

**Reply by gwijnja** [reply]
Yes, but that only works for one set. If the selected parts (accidentally) contain one extra part, then the set won't show up anymore. For example:
1) start with an empty list
2) add a set
3) part out the set
4) ctrl+a: the set will be shown on the left hand side. Great!
5) add one extra random part to the list
6) ctrl+a again: the set will not be shown anymore. But the set matching feature description implies that it "finds sets that you could build from a given list of items".

So if you bought a box of random lego parts, without instructions, and you are trying to find out what sets it contains, then the "set matching" feature sound like exactly what you need: just ctrl+a on all the parts in the list, and find out what sets you can build with them.  That's what the feature description sounds like: "_given a list of items, Brickstore can tell you which sets you could build from them._"

Ideally, it would even show sets that are almost complete, but that are missing a few parts. So it could list sets that are "95% complete" or so. Because that's very common if you buy a lot of random parts. I'm thinking about writing an application that does this.

**Comment by rgriebl** [comment]
Set matching was a planned feature, but the implementation didn't work out: see https://github.com/rgriebl/brickstore/issues/40

I totally forgot that I had this feature in the app description...

**Reply by gwijnja** [reply]
Ah, then I can stop looking. Thanks.

Yes it's not a trivial problem with an obvious solution. There are many different approaches and considerations, with many possible outcomes. I see you were considering implementing the knapsack problem to solve it. I'm thinking more about a tool that gives hints about possible sets, based on a percentage of a set that is complete (i.e. more than 50% complete), but also taking into account that some parts appear in only some sets. For example if a part exists in only one set, but you only have 10% of that set in your inventory, then I would still want to know about it. But what if a part exists in only two sets? That's where overlap with other sets begins, and then it gets tricky. We'll see. :)

**Reply by ZZJHONS** [reply]
@gwijnja You can create the list in BrickStore and then export it to Rebrickable where they have that function.

**Reply by gwijnja** [reply]
Ooh that's awesome, looks great indeed. This saves me a lot of time. Thank you!

**Comment by StoneyBricks** [comment]
I understand now. 

I haven’t figured that out yet. 

I did see something that you described on Brickset. If you load your “inventory”, it will show which sets can be built and what the percentage of parts you have to complete the set. But, being honest, I haven’t used that all all. I just tried it once with a small selection of parts. 

Perhaps there is a way to do it in Brickstore and hopefully an administrator will reply. 

Regards
John Wybenga
